stac_fastapi/sfeos_helpers/stac_fastapi/sfeos_helpers/queryables.py

```diff
@@ -0,0 +1,146 @@
+        self._cache: Dict[str, List[str]] = {}
+        self._all_queryables: Set[str] = set()
+        self._last_updated: float = 0
+        self._lock = asyncio.Lock()
+        self.validation_enabled: bool = False
+        self.cache_ttl: int = 3600  # How often to refresh cache (in seconds)
+        self.excluded_queryables: Set[str] = set()
+        self.reload_settings()
+
+    def reload_settings(self):
+        """Reload settings from environment variables."""
+        self.validation_enabled = (
+            os.getenv("VALIDATE_QUERYABLES", "false").lower() == "true"
+        )
+        self.cache_ttl = int(os.getenv("QUERYABLES_CACHE_TTL", "3600"))
+
+        excluded = os.getenv("EXCLUDED_FROM_QUERYABLES", "")
+        self.excluded_queryables = set()
+        if excluded:
+            for field in excluded.split(","):
+                field = field.strip()
+                if field:
+                    # Remove 'properties.' prefix if present
+                    if field.startswith("properties."):
+                        field = field[11:]
+                    self.excluded_queryables.add(field)
+
+    async def _update_cache(self):
+        """Update the cache with the latest queryables from the database."""
+        if not self.validation_enabled:
+            return
+
+        async with self._lock:
+            if (time.time() - self._last_updated < self.cache_ttl) and self._cache:
+                return
+
+            queryables_mapping = await self._db_logic.get_queryables_mapping()
+            all_queryables_set = set(queryables_mapping.keys())
+
+            if self.excluded_queryables:
+                all_queryables_set = all_queryables_set - self.excluded_queryables
+
+            self._all_queryables = all_queryables_set
+
+            self._cache = {"*": list(all_queryables_set)}
+            self._last_updated = time.time()
+
+    async def get_all_queryables(self) -> Set[str]:
+        """
+        Return a set of all queryable attributes across all collections.
+
+        This method will update the cache if it's stale or has been cleared.
+        """
+        if not self.validation_enabled:
+            return set()
+
+        if (time.time() - self._last_updated >= self.cache_ttl) or not self._cache:
+            await self._update_cache()
+        return self._all_queryables
+
+    async def validate(self, fields: Set[str]) -> None:
+        """
+        Validate if the provided fields are queryable.
+
+        Raises HTTPException if invalid fields are found.
+        """
+        if not self.validation_enabled:
+            return
+
+        allowed_fields = await self.get_all_queryables()
+        invalid_fields = fields - allowed_fields
+        if invalid_fields:
+            raise HTTPException(
+                status_code=400,
+                detail=f"Invalid query fields: {', '.join(invalid_fields)}.",
+            )
+
+
+_queryables_cache_instance: Optional[QueryablesCache] = None
+
```

Comment:
Please refactor the module-level global variable _queryables_cache_instance in queryables.py. Using global mutable state in an async application is not thread-safe and makes testing fragile (as seen by the need to manually reset it in the test fixtures). Please remove the global variable. Instead, initialize the QueryablesCache within the lifespan context manager in stac_fastapi/elasticsearch/app.py and attach it to app.state.